packages/block-library/src/media-text/edit.js

```diff
@@ -50,6 +104,67 @@ class MediaTextEdit extends Component {
 		this.state = {
 			mediaWidth: null,
 		};
+		this.getLinkDestinations = this.getLinkDestinations.bind( this );
+		this.onSetHref = this.onSetHref.bind( this );
+		this.onSetNewTab = this.onSetNewTab.bind( this );
+		this.onSetLinkRel = this.onSetLinkRel.bind( this );
+		this.onSetLinkClass = this.onSetLinkClass.bind( this );
+	}
+
+	getLinkDestinations() {
```

Comment:
Would it be possible to remove the need to have functions getLinkDestinations, onSetHref, onSetNewTab, onSetLinkRel, onSetLinkClass?


I guess if ImageURLInputUI receives mediaUrl, and mediaLink URL we can easily compute getLinkDestinations inside the component.
Then we can make ImageURLInputUI receive a onChangeLink prop and this prop is a function that may receive an object with props: href, linkDestination, rel, linkTarget. Doing this all the link changing and utils may be inside the component.

Reply:
Hey @jorgefilipecosta I have moved the functionality to the new component as indeed all these functions served only the ImageURLInputUI component.

Reply:
There appears to be a bug that does not update the label of the link once you change to a custom link destination ... on it.

Reply:
It is either fixed or I can't reproduce it anymore ...